libraries/src/Application/CMSApplication.php

```diff
@@ -904,40 +904,72 @@ public function login($credentials, $options = [])
             $authorisations = $authenticate->authorise($response, $options);
             $denied_states  = Authentication::STATUS_EXPIRED | Authentication::STATUS_DENIED;
 
-            foreach ($authorisations as $authorisation) {
-                if ((int) $authorisation->status & $denied_states) {
-                    // Trigger onUserAuthorisationFailure Event.
-                    $dispatcher->dispatch('onUserAuthorisationFailure', new AuthorisationFailureEvent('onUserAuthorisationFailure', [
-                        'subject' => (array) $authorisation,
-                        'options' => $options,
-                    ]));
-
-                    // If silent is set, just return false.
-                    if (isset($options['silent']) && $options['silent']) {
-                        return false;
-                    }
+            // Validate that $authorisations is an array before iterating
+            if (!is_array($authorisations)) {
+                // Log the issue for debugging purposes
+                \JLog::add(
+                    sprintf(
+                        'Invalid data type for $authorisations in %s on line %d. Expected array, got %s.',
+                        __FILE__,
+                        __LINE__,
+                        gettype($authorisations)
```

Comment:
```suggestion
            if (!\is_array($authorisations)) {
                // Log the issue for debugging purposes
                \JLog::add(
                    \sprintf(
                        'Invalid data type for $authorisations in %s on line %d. Expected array, got %s.',
                        __FILE__,
                        __LINE__,
                        \gettype($authorisations)
```
PHPCS global namespace (1)